app/src/main/java/edu/wpi/first/shuffleboard/app/json/WidgetSaver.java

```diff
@@ -31,7 +36,11 @@ public Widget deserialize(JsonElement json, JsonDeserializationContext context)
             .orElseThrow(() -> new JsonParseException("No widget found for " + type)).get();
 
     String source = obj.get("_source").getAsString();
-    widget.setSource(Sources.getDefault().forUri(source));
+    try {
+      widget.setSource(Sources.getDefault().forUri(source));
+    } catch (RuntimeException e) {
```

Comment:
This is very broad... what kind of exceptions are you expecting to be thrown, and under what circumstances?

Reply:
hmm it looks like this API has changed since I wrote this code. forUri used to throw a runtime exception if the protocol wasn't registered.